dpe/src/dpe_instance.rs

```diff
@@ -151,27 +152,27 @@ impl DpeInstance {
         handle: &ContextHandle,
         locality: u32,
     ) -> Result<usize, DpeErrorCode> {
-        let mut valid_handles = self
+        let mut valid_localities = self
             .contexts
             .iter()
             .enumerate()
             .filter(|(_, context)| {
-                context.state == ContextState::Active && &context.handle == handle
+                context.state == ContextState::Active && context.locality == locality
             })
             .peekable();
-        if valid_handles.peek().is_none() {
-            return Err(DpeErrorCode::InvalidHandle);
+        if valid_localities.peek().is_none() {
+            return Err(DpeErrorCode::InvalidLocality);
         }
-        let mut valid_handles_and_localities = valid_handles
-            .filter(|(_, context)| context.locality == locality)
+        let mut valid_handles_and_localities = valid_localities
+            .filter(|(_, context)| constant_time_eq(&context.handle.0, &handle.0))
```

Comment:
Just for curiosity's sake, did you switch the order here to make it harder for an attacker to see how many total active contexts there are? The most they could see is how many active contexts are in their locality?

Reply:
I'm not totally sure about the reasoning, it was just one of NCC's recommendations.

Reply:
This does reduce the number of handles that you need to do a handle comparison on, so I think it's more of a performance optimization than a security optimization.